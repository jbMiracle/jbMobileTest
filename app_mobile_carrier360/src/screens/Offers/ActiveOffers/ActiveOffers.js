import React, { Component } from 'react';
import { View } from 'react-native';
import { Text, Button } from 'native-base';
import C360Colors from '../../../themes/360Colors';

class ActiveOffers extends Component {
  render() {
    return (
      <View testID='offers' style={ { flex: 1, justifyContent: 'center', alignItems: 'center' } }>
        <Text>Active Offers</Text>
        <Button
          style={ {
              backgroundColor: C360Colors.Yellow500, padding: 0, marginTop: 15, alignSelf: 'center',
          } }
          onPress={ () => this.props.navigation.navigate('Offer Details', { msg: 'Here are the details...' }) }
        >
          <Text style={ { color: C360Colors.Blue600, fontSize: 12, fontWeight: 'bold' } }>Show Offer Details</Text>
        </Button>
      </View>
    );
  }
}

export default ActiveOffers;
