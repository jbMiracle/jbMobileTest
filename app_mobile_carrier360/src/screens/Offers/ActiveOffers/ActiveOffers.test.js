import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ActiveOffers from '../ActiveOffers/ActiveOffers';

it('can render an empty component', () => {
  const renderer = new ShallowRenderer();

  renderer.render(<ActiveOffers />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
