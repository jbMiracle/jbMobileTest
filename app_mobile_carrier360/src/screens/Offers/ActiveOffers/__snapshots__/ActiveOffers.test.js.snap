// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render an empty component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="offers"
>
  <Styled(Text)
    virtual={undefined}
  >
    Active Offers
  </Styled(Text)>
  <Styled(Button)
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFDB00",
        "marginTop": 15,
        "padding": 0,
      }
    }
    virtual={undefined}
  >
    <Styled(Text)
      style={
        Object {
          "color": "#005F8E",
          "fontSize": 12,
          "fontWeight": "bold",
        }
      }
      virtual={undefined}
    >
      Show Offer Details
    </Styled(Text)>
  </Styled(Button)>
</View>
`;
